keanu-project/src/main/java/io/improbable/keanu/vertices/generic/probabilistic/discrete/SelectVertex.java

```diff
@@ -3,26 +3,25 @@
 import io.improbable.keanu.vertices.dbl.DoubleVertex;
 import io.improbable.keanu.vertices.generic.probabilistic.Probabilistic;
 
-import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Random;
 
 public class SelectVertex<T> extends Probabilistic<T> {
 
-    private final LinkedHashMap<T, DoubleVertex> selectableValues;
+    private final Map<T, DoubleVertex> selectableValues;
```

Comment:
I think ordering is actually important here. There should actually be a test that checks this if there isn't.

Reply:
It might just be that the iteration order needs to be the same. So maybe we can make a defensive copy.

Reply:
Okay, I've reverted it back to `LinkedHashMap` to preserve the ordering